### PR TITLE
Fix error without filterableColumns. Added logger

### LINKED ISSUE #380: Filtering with undefined `filterableColumns` throws cryptic error
Using `paginate` without `filterableColumns` in the config throws (with `in` as operator and `id` as column):

```
Can't use operator 'in' to find 'id' in undefined
```

### PR DESCRIPTION
**Pull request by Helveg**
Fixes #380 and adds a logger that reports the error reasons on debug level. Added troubleshooting section in README

### CODE REVIEW

**Comment by ppetzold** [comment]
Sweet, looks good to me 🌟 

CI fails due to prettier check. You can run `npm run format` to fix.

**Comment by Helveg** [comment]
I knew editing it in GitHub would be too good to be true 🥲 Done!

File: src/paginate.ts [hunk 1]
```diff
@@ -119,6 +121,11 @@ export function getFilterTokens(raw: string): string[] {
 
 function parseFilter<T>(query: PaginateQuery, config: PaginateConfig<T>) {
     const filter: { [columnName: string]: FindOperator<string> } = {}
+    let filterableColumns = config.filterableColumns
```
Comment (ppetzold) [inline src/paginate.ts:124]: just noticed now, variable is unused. please remove if not needed

**Review by ppetzold — CHANGES_REQUESTED** [review]

File: src/paginate.ts (hunk 1 above)
Comment (Helveg) [reply src/paginate.ts:124]: Oh sorry, I forgot to replace a later occurence of `config.filterableColumns` with just `filterableColumns`

**Pull request merged**